mne/chpi.py

```diff
@@ -84,24 +92,87 @@ def get_chpi_positions(raw, t_step=None, return_quat=False, verbose=None):
     else:
         if not isinstance(raw, string_types):
             raise TypeError('raw must be an instance of Raw or string')
-        if not op.isfile(raw):
-            raise IOError('File "%s" does not exist' % raw)
         if t_step is not None:
             raise ValueError('t_step must be None if processing a log')
-        data = np.loadtxt(raw, skiprows=1)  # first line is header, skip it
-    out = _quats_to_trans_rot_t(data)
+        data = read_head_quats(raw)
+    out = head_quats_to_trans_rot_t(data)
     if return_quat:
         out = out + (data[:, 1:4],)
     return out
 
 
-def _quats_to_trans_rot_t(quats):
+def read_head_quats(fname):
+    """Read MaxFilter-formatted head position parameters
+
+    Parameters
+    ----------
+    fname : str
+        The filename to read. This can be produced by e.g.,
+        ``maxfilter -headpos <name>.pos``.
+
+    Returns
+    -------
+    pos : array, shape (N, 10)
+        The position parameters.
+
+    See Also
+    --------
+    write_head_quats
+    head_quats_to_trans_rot_t
+
+    Notes
+    -----
+    .. versionadded:: 0.12
+    """
+    if not isinstance(fname, string_types):
+        raise TypeError('fname must be str, not %s' % type(fname))
+    if not op.isfile(fname):
+        raise IOError('File "%s" does not exist' % fname)
+    data = np.loadtxt(fname, skiprows=1)  # first line is header, skip it
+    data.shape = (-1, 10)  # ensure it's the right size even if empty
+    return data
+
+
+def write_head_quats(fname, pos):
+    """Write MaxFilter-formatted head position parameters
+
+    Parameters
+    ----------
+    fname : str
+        The filename to write.
+    pos : array, shape (N, 10)
+        The position parameters.
+
+    See Also
+    --------
+    read_head_quats
+    head_quats_to_trans_rot_t
+
+    Notes
+    -----
+    .. versionadded:: 0.12
+    """
+    if not isinstance(fname, string_types):
+        raise TypeError('fname must be a string')
```

Comment:
Any reason not to match above?
`raise TypeError('fname must be str, not %s' % type(fname))`